### PR TITLE
Fixes #1 : Toolbar should be displayed even without <head> in HTML5

### LINKED ISSUE #1: In HTML5, the <head> element can be omitted. The zend-developer-tools css requires it.
- [x] I was not able to find an [open](https://github.com/zendframework/zend-developer-tools/issues?q=is%3Aopen) or [closed](https://github.com/zendframework/zend-developer-tools/issues?q=is%3Aclosed) issue matching what I'm seeing.
 - [x] This is not a question. (Questions should be asked on [chat](https://zendframework.slack.com/) ([Signup here](https://zendframework-slack.herokuapp.com/)) or our [forums](https://discourse.zendframework.com/).)

Provide a narrative description of what you are trying to accomplish.

### Code to reproduce the issue

In HTML5, the &lt;head&gt; element can be omitted. The following code will validate as HTML5:
```html
<!DOCTYPE html>
<html>
<title>Title of the document</title>

<body>
<h1>This is a heading</h1>
<p>This is a paragraph.</p>
</body>

</html>
```

However, ZendDeveloperTools\Listener->injectToolbar() requires it to work
```php
$injected = preg_replace('/<\/head>/i', $style . "\n</head>", $injected, 1);
```

### Expected results

Styles should be displayed in the html document even if there is no &lt;head&gt; html element when the document is html 5 document.

### Actual results

html document does not include zend-developer-tools required css, if &lt;head&gt; element is missing from the layout.

---

Originally posted by @villetikkanen at https://github.com/zendframework/zend-developer-tools/issues/255

### PR DESCRIPTION
**Pull request by samsonasik**
|    Q          |   A
|-------------- | ------
| Bugfix        | yes

### Description

Fixes #1 : Toolbar should be displayed even without <head> in HTML5

### CODE REVIEW

**Comment by samsonasik** [comment]
Make toolbar still showing with and without head:

**With head:**

Screen Shot 2020-09-22 at 10 39 57 AM


**Without head:**

Screen Shot 2020-09-22 at 10 40 18 AM

**Comment by samsonasik** [comment]
closing reopen to restart travis build

**Comment by samsonasik** [comment]
travis actually green

File: src/Listener/ToolbarListener.php [hunk 1]
```diff
@@ -135,7 +135,8 @@ protected function injectToolbar(ProfilerEvent $event)
             $response->getBody(),
             1
         );
-        $injected = preg_replace('/<\/head>/i', $style . "\n</head>", $injected, 1);
+        $prepend = preg_match('/<\/head>/i', $injected) ? 'head' : 'body';
+        $injected = preg_replace('/<\/' . $prepend . '>/i', $style . "\n</$prepend>", $injected, 1);
```
Comment (froschdesign) [inline src/Listener/ToolbarListener.php]: And what about the case when no `body` element exists?

Comment (samsonasik) [reply src/Listener/ToolbarListener.php]: do you have suggestion?

Comment (froschdesign) [reply src/Listener/ToolbarListener.php]: Check always for `head` and `body`. If both does not present, add the HTML of the toolbar to the end.

Comment (samsonasik) [reply src/Listener/ToolbarListener.php]: implemented, please verify

File: src/Listener/ToolbarListener.php [hunk 2]
```diff
@@ -129,14 +129,26 @@ protected function injectToolbar(ProfilerEvent $event)
...
+            );
+
+            $prepend = stripos($body, '<!doctype html>') === 0
+                ? (preg_match('/<\/head>/i', $injected) ? 'head' : 'body')
+                : 'body';
+
+            $injected = preg_replace('/<\/' . $prepend . '>/i', $style . "\n</$prepend>", $injected, 1);
+        } else {
+            if (stripos($body, '<!doctype html>') === 0) {
+                $injected = preg_replace('/<\/html>/i', $style . $toolbar . $script . "\n</html>", $body, 1);
```
Comment (froschdesign) [inline src/Listener/ToolbarListener.php]: Please note that the `html` element may not be available:

```html
<!doctype html><title>Minimal</title>
```

Comment (samsonasik) [reply src/Listener/ToolbarListener.php]: fixed.

**Comment by samsonasik** [comment]
All handled now

**Comment by samsonasik** [comment]
@froschdesign is there anything I can do to get it merged? Thank you.

**Comment by samsonasik** [comment]
I will merge it tomorrow if no more review ;)

**Review by froschdesign — CHANGES_REQUESTED** [review]
Some unit tests should be added to cover all types of HTML documents.

**Comment by samsonasik** [comment]
test has been added.

**Comment by samsonasik** [comment]
All green :tada:

**Pull request merged**